internal/collector/event_test.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2020 Red Hat, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collector
+
+import (
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestBeforeLatestEvent(t *testing.T) {
+	now := time.Date(1970, time.January, 1, 0, 0, 0, 0, time.UTC)
+	older := now.Add(-time.Minute)
+	newer := now.Add(time.Minute)
+
+	olderEvent := &v1.Event{}
+	olderEvent.CreationTimestamp = metav1.NewTime(older)
+
+	scenarios := []struct {
+		Desc     string
+		Event    *v1.Event
+		Expected bool
+	}{
+		{
+			Desc: "newer EventTime and other older time instants",
+			Event: &v1.Event{
+				FirstTimestamp: metav1.NewTime(older),
+				LastTimestamp:  metav1.NewTime(older),
+				EventTime:      metav1.NewMicroTime(newer),
+				Series:         &v1.EventSeries{LastObservedTime: metav1.NewMicroTime(older)},
+			},
+			Expected: true,
+		},
+		{
+			Desc: "only newer LastObservedTime",
+			Event: &v1.Event{
+				Series: &v1.EventSeries{LastObservedTime: metav1.NewMicroTime(newer)},
+			},
+			Expected: true,
+		},
+		{
+			Desc: "only older time instants",
+			Event: &v1.Event{
+				FirstTimestamp: metav1.NewTime(older),
```

Comment:
Could we have a case for:
- newer, older, now
- newer, older, older
- etc?

We just want to cover all cases, unless you think these are the only ones that we need to cover?

Reply:
I don't think it is meaningful to cover all cases tbh. There are a lot of scenarios that we could provide to this function but in practice, there will only be 4-5 scenarios.
For this test, I did not provide real Event scenarios as those will most likely be tested in e2e tests. However, the scenario I came with were: "one newer timestamp", "only one newer timestamp field provided", "no newer timestamp", "only one older timestamp field provided". What do you think?

Reply:
> function but in practice, there will only be 4-5 scenarios.

I don't see 4-5 scenarios here. It's not urgent to add them, but nice to add have more of them. It's not blocking this PR.